apps/backend/src/core/links/links.controller.ts

```diff
@@ -48,6 +49,8 @@ export class LinksController {
       throw new UnauthorizedException();
     }
 
+    await this.cacheService.del(link.key);
+
     return this.linksService.delete(id);
```

Comment:
I think it's a good approach to use Promise.all to delete it from cache and database concurrently.

```suggestion
    await Promise.all([
        this.cacheService.del(link.key),
        this.linksService.delete(id)
    ]);

    return link;
```